### PR TITLE
Add trigonometric functions

### LINKED ISSUE #854: Document trigonometric functions
### Related code PR
pr1: https://github.com/risingwavelabs/risingwave/pull/8838

pr2: https://github.com/risingwavelabs/risingwave/pull/8918
issue2: https://github.com/risingwavelabs/risingwave/issues/9675

pr3: https://github.com/risingwavelabs/risingwave/pull/9203

pr4: https://github.com/risingwavelabs/risingwave/pull/9259

### Which part(s) of the docs might be affected or should be updated? And how?

https://www.risingwave.dev/docs/docs/upcoming/sql-function-mathematical/#trigonometric-functions

### Reference

See `e2e_test/batch/functions/trigonometric_funcs.slt.part` in the code repo.

### PR DESCRIPTION
**Pull request by bernscode**
Add trigonometric functions.

## Info
- **Description**: 
Add trigonometric functions.

- **Preview**: 
https://pr-935.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-function-mathematical/

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/8838
https://github.com/risingwavelabs/risingwave/pull/8918
https://github.com/risingwavelabs/risingwave/pull/9203
https://github.com/risingwavelabs/risingwave/pull/9259

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/854

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-mathematical.md [hunk 1]
```diff
@@ -36,4 +36,22 @@ title: Mathematical functions and operators
...
\ No newline at end of file
+| sin ( *radians* ) → *sine* | Returns the trigonometric sine (in double precision) of an angle measured in radians (in double precision). | sin(1) → 0.8414709848078965 |
+| cos ( *radians* ) → *cosine* | Returns the trigonometric cosine (in double precision) of an angle measured in radians (in double precision). | cos(1) → 0.5403023058681398 |
+| tan ( *radians* ) → *tangent* | Returns the trigonometric tangent (in double precision) of an angle measured in radians (in double precision). | tan(1) → 1.5574077246549021 |
+| cot ( *radians* ) → *cotangent* | Returns the trigonometric cotangent (in double precision) of an angle measured in radians (in double precision). | cot(1) → 0.6420926159343308 |
+| asin ( *input_value* ) → *radians* | Returns the inverse sine (in radians and double precision) of a given value (in double precision). | asin(0.5) → 0.5235987755982989 |
+| acos ( *input_value* ) → *radians* | Returns the inverse cosine (in radians and double precision) of a given value (in double precision). | acos(0.5) → 1.0471975511965979 |
```
Comment (CAJan93) [inline docs/sql/functions-operators/sql-function-mathematical.md]:
```suggestion
| acos ( *input_value* ) → *radians* | Returns the inverse cosine (in radians and double precision) of a given value (in double precision). | acos(0.5) → 1.0471975511965976 |
```

File: docs/sql/functions-operators/sql-function-mathematical.md [hunk 2]
```diff
@@ -36,4 +36,22 @@ title: Mathematical functions and operators
...
+| cosh ( *input_value* ) → *hyperbolic_cosine* | Returns the hyperbolic cosine (in double precision) of a given value (in double precision). | cosh(1.0) → 1.5430806348152437 |
+| tanh ( *input_value* ) → *hyperbolic_tangent* | Returns the hyperbolic tangent (in double precision) of a given value (in double precision). | tanh(1.0) → 0.7615941559557649 |
+| coth ( *input_value* ) → *hyperbolic_cotangent* | Returns the hyperbolic cotangent (in double precision) of a given value (in double precision). | coth(2) → 1.0373147207275481 |
+| asinh ( *input_value* ) → *inverse_hyperbolic_sine* | Returns the inverse hyperbolic sine (in double precision) of a given value (in double precision). | asinh(1.0) → 0.881373587019543 |
+| acosh ( *input_value* ) → *inverse_hyperbolic_cosine* | Returns the inverse hyperbolic cosine (in double precision) of a given value (in double precision). | acosh(2.0) → 1.3169578969248166 |
+| atanh ( *input_value* ) → *inverse_hyperbolic_tangent* | Returns the inverse hyperbolic tangent (in double precision) of a given value (in double precision). | atanh(0.5) → 0.5493061443340548 |
```
Comment (CAJan93) [inline docs/sql/functions-operators/sql-function-mathematical.md]:
```suggestion
| atanh ( *input_value* ) → *inverse_hyperbolic_tangent* | Returns the inverse hyperbolic tangent (in double precision) of a given value (in double precision). | atanh(0.5) → 0.5493061443340549 |
```

File: docs/sql/functions-operators/sql-function-mathematical.md [hunk 3]
```diff
@@ -36,4 +36,22 @@ title: Mathematical functions and operators
...
+| sind ( *degrees* ) → *sine* | Returns the trigonometric sine (in double precision) of an angle measured in degrees (in double precision). | sind(15) → 0.2588190451025207 |
```
Comment (CAJan93) [inline docs/sql/functions-operators/sql-function-mathematical.md]:
```suggestion
| sind ( *degrees* ) → *sine* | Returns the trigonometric sine (in double precision) of an angle measured in degrees (in double precision). | sind(15) → 0.2588190451025208 |
```

File: docs/sql/functions-operators/sql-function-mathematical.md [hunk 4]
```diff
@@ -36,4 +36,22 @@ title: Mathematical functions and operators
...
+| cosd ( *degrees* ) → *cosine* | Returns the trigonometric cosine (in double precision) of an angle measured in degrees (in double precision). | cosd(15) → 0.9659258262890683 |
+| tand ( *degrees* ) → *tangent* | Returns the trigonometric tangent (in double precision) of an angle measured in degrees (in double precision). | tand(15) → 0.2679491924311227|
```
Comment (CAJan93) [inline docs/sql/functions-operators/sql-function-mathematical.md]:
```suggestion
| tand ( *degrees* ) → *tangent* | Returns the trigonometric tangent (in double precision) of an angle measured in degrees (in double precision). | tand(15) → 0.26794919243112275 |
```

**Review by CAJan93 — APPROVED** [review]
LGTM. Thanks for adding the docs 👍 No idea if your numbers are closer to the truth as compared to mine. I ran them through RW. Let's use the numbers we actually output.

**Review by CAJan93 — APPROVED** [review]
LGTM. Thanks for adding the docs 👍 No idea if your numbers are closer to the truth as compared to mine. I ran them through RW. Let's use the numbers we actually output.

**Comment by bernscode** [comment]
@CAJan93 I've made the suggested changes to the latest commit. Thanks!

**Review by hengm3467 — APPROVED** [review]
LGTM. Not related to this PR specifically, but maybe we can think about how to improve the presentation of function docs.

**Pull request merged**